zookeeper-server/src/test/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProviderTest.java

```diff
@@ -84,7 +86,8 @@ public void setUp() throws Exception {
     LOG.info("Starting Zk...");
     zks = new ZooKeeperServer(testBaseDir, testBaseDir, 3000);
     final int PORT = Integer.parseInt(HOSTPORT.split(":")[1]);
-    serverCnxnFactory = ServerCnxnFactory.createFactory(PORT, -1);
+    serverCnxnFactory = new TestNIOServerCnxnFactory();
+    serverCnxnFactory.configure(new InetSocketAddress(PORT), -1, -1);
```

Comment:
Is there a reason we're using NIOServerCnxnFactory? Ideally, we should test for both NIO and Netty based implementations.